spec/index.md

```diff
@@ -5078,6 +5082,61 @@ S with
 
 ```
 
+#### IC Management Canister: List snapshots
+
+Only the controllers of the given canister can get a list of the existing snapshots.
+
+```html
+
+S.messages = Older_messages · CallMessage M · Younger_messages
+(M.queue = Unordered) or (∀ msg ∈ Older_messages. msg.queue ≠ M.queue)
+M.method_name = 'list_snapshots'
+M.arg = candid(A)
+S.snapshots = Other_snapshots ∪ S.snapshots[A.snapshot_id]
+
+```
+
+State after
+
+```html
+
+S with
+    messages = Older_messages · Younger_messages ·
+      ResponseMessage {
+        origin = M.origin
+        response = candid(dom(S.snapshots[A.canister_id]))
+      }
+
+```
+#### IC Management Canister: Delete snapshot
+
+A snapshot may be deleted only by the controllers of the canister for which the snapshot was taken.
+
+```html
+
+S.messages = Older_messages · CallMessage M · Younger_messages
+(M.queue = Unordered) or (∀ msg ∈ Older_messages. msg.queue ≠ M.queue)
+M.method_name = 'delete_snapshot'
+M.arg = candid(A)
+M.caller ∈ S.controllers[S.snapshots[A.snapshot_id].canister_id]
+S.snapshots = Other_snapshots ∪ S.snapshots[A.snapshot_id]
```

Comment:
I'd drop this line and define
```
S.snapshots[A.snapshot_id] = (deleted)
```
in the state after (see canister deletion for reference)